src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -430,6 +465,39 @@ namespace winrt::TerminalApp::implementation
                 tab->_RecalculateAndApplyTabColor();
             }
         });
+
+        control.SetTaskbarProgress([weakThis](auto&&, auto&&) {
+            // Check if Tab's lifetime has expired
+            if (auto tab{ weakThis.get() })
+            {
+                // The progress of the control changed, but not necessarily the progress of the tab.
+                // Set the tab's progress ring to the active pane's progress
+                if (!tab->GetActiveTerminalControl().Settings().DisableProgressRing())

```

Comment:
nit: we could probably cache `tab->GetActiveTerminalControl()` in a local somewhere at the top of this lambda